hal/halx86/generic/usage.c

```diff
@@ -32,71 +32,18 @@ USHORT HalpComPortIrqMapping[5][2] =
 
 ADDRESS_USAGE HalpComIoSpace =
 {
-    NULL, CmResourceTypePort, IDT_INTERNAL,
+    NULL, CmResourceTypePort, IDT_DEVICE,
     {
-        {0x2F8,   0x8},     /* COM 1 */
+        {0x2F8,   0x8},     /* COM 2 */
```

Comment:
I confirm this change (see e.g. boot/freeldr/freeldr/lib/comm/rs232.c line 43; boot/freeldr/freeldr/ntldr/headless.c line 125+; ntoskrnl/inbv/inbvport.c line 136+).
And I also wonder, why does the ComIoSpace only contains info for the COM2 and not the COM1?

Reply:
HalpComIoSpace is used if KD is using a COM port
![HalpComIoSpace](https://user-images.githubusercontent.com/17290914/111365163-0e645f80-868a-11eb-9412-36f8418b3c40.png)

In the time of NT5.1 | 2, computers motherboards have had a back connector for COM ports
![COM-connectors](https://user-images.githubusercontent.com/17290914/111365514-7ca92200-868a-11eb-998a-faae87f41b27.jpg)

In those days, the COM1 was usually used for modems
![U S  Robotics V  Everything Modem Model 34538 Courier 56K Business](https://user-images.githubusercontent.com/17290914/111365981-dd385f00-868a-11eb-914d-a0e1b59ab191.png)

I think that in order not to touch the COM1 port, the null-modem cable has been connected to the COM2 port by default.